src/StoreKeeper/WooCommerce/B2C/PaymentGateway/PaymentGateway.php

```diff
@@ -272,9 +299,75 @@ public function createRefundPayment($orderId, $refundId)
                     throw new \Exception('Order export task was not created');
                 }
             }
+        } catch (GeneralException $generalException) {
+            if ('Only invoiced orders can be refunded' === $generalException->getMessage()) {
+                self::createRefundAsPayment($orderId, $refundId, $refundAmount);
+                self::scheduleRefundTask($orderId, $refundId, $refundAmount);
+            }
```

Comment:
what with other errors why do we ignore them?

Reply:
This is the error I got when the `refundAllOrderItems` was called and then a new refund was made again (for staggered refunds). I'm just worried the client's shops will be spammed by refund tasks if other issues will be caught. What is your insight in this?